packages/openapi-generator/src/file-serializer/api-file.ts

```diff
@@ -16,25 +21,29 @@ import type {
  * Serialize an API representation to a string representing the resulting API file.
  * @param api - Representation of an API.
  * @param serviceName - Service name for which the API is created.
+ * @param options - Options to configure the file creation.
+ * @param basePath - Base path for the API obtained from optionsPerService.
  * @returns The serialized API file contents.
  * @internal
  */
 export function apiFile(
   api: OpenApiApi,
   serviceName: string,
-  options?: CreateFileOptions
+  options?: CreateFileOptions,
+  basePath?: string
 ): string {
   const imports = serializeImports(getImports(api, options));
   const apiDoc = apiDocumentation(api, serviceName);
+  const defaultBasePath = basePath ? removeSlashes(basePath) : undefined;
```

Comment:
[req] Why are we removing slashes? Shouldn't path always start with '/'? If user provides a path ending with '/', it should be respected (may not be intentional but still a valid path). https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname

Reply:
[req] I feel we should not be sanitizing paths the user set in the generated code. 
We can remove slashes later when we are building the final url, just like we do in the OData request builder.

Reply:
It's just that we are sanitising it anyway later before execution so, why not make it evident on the generated code too?

Reply:
True, I also check OData before and it is not sanitized. I would not remove the slashes and the user should know what to provide.

Reply:
Done.